### PR TITLE
Remove Sourcegraph Cloud option

### LINKED ISSUE #5508: Remove Sourcegraph Cloud option

### PR DESCRIPTION
**Pull request by zlonko**
This closes #5508. It removes the option for Sourcegraph Cloud from "Get Started" and adds the appropriate redirects and links.

6/23/22: Requirements not finalized

### Changelog
- Updates "Get Started" CTA on homepage and nav `/get-started/self-hosted`
- Redirects `/get-started` to `/get-started/self-hosted`
- Removes 'Deployment options' on `/get-started/self-hosted`
- Updates the `SelfHostedSection.tsx` component's CTA copy to "Learn more about self-hosted" and link to `/get-started/self-hosted`

### Test
1. Ensure prettier has standardized the proposed changes.
2. Please test links in nav and homepage
3. Please test redirect when visiting `/get-started`

### CODE REVIEW

**Comment by zlonko** [comment]
@elzannewentzel The redirects are a good short-term fix, but we may consider repurposing the design of `/get-started` as well. This may be more clear to users in the long-term and would simplify the structure of this directory.

**Comment by zlonko** [comment]
Sought stakeholder approval before code review: Andy and Fabiana approve, and Alex will review when he returns on Monday (6/27).

File: netlify.toml
Comment (st0nebraker) [inline netlify.toml:150]: Can we add a redirect for `/get-started/cloud` too? I notice this page is still accessible. Interestingly, the back `Deployment Options` button direct a user to `/get-started` from this page & it doesn't redirect to `/get-started/self-hosted` 🤔  A redirect for `/get-started/cloud` should cover this though
image

**Review by bretthayes — APPROVED** [review]
Made a quick [callout](https://sourcegraph.slack.com/archives/C02PSJF0QLU/p1656345223443539?thread_ts=1655415090.407949&cid=C02PSJF0QLU) on potentially updating copy but LGTM so far!

**Review by st0nebraker — CHANGES_REQUESTED** [review]
Looks good! One request below. Sorry if I missed this, but the idea is that we'll be using the Cloud deployment page again later, that's why we're not removing it completely?

**Review by bretthayes — CHANGES_REQUESTED** [review]
As per this [comment](https://sourcegraph.slack.com/archives/C02PSJF0QLU/p1656345805845899?thread_ts=1655415090.407949&cid=C02PSJF0QLU), could you remove the `<MostPopularBadge />` and change the copy of the `Best For` heading to `Designed For`?

File: netlify.toml
Comment (zlonko) [reply netlify.toml:150]: This is a great catch, Becca. I have added the redirect here: a89bb93. 

I also posed the question of removing the Cloud page to Andy, Alex, and Fabiana and will return with a response. As you noted, it makes sense to me that we would remove the page entirely if "Sourcegraph Cloud" is being phased out of About.

File: src/pages/get-started/index.tsx
Comment (zlonko) [inline src/pages/get-started/index.tsx:19]: @bretthayes Highlighting that the update from "Best For" -> "Designed For" also resulted in the name change to `DesignedForTitle`. ef9ef01

**Review by st0nebraker — APPROVED** [review]
LGTM!

File: netlify.toml
Comment (zlonko) [reply netlify.toml:150]: Per this [comment](https://sourcegraph.slack.com/archives/C03M3338LP5/p1656350207350549?thread_ts=1656349394.483309&cid=C03M3338LP5), it looks like we will retain "Get Started / Cloud", and Alex is redesigning the user flow. Let's keep this page so that it can be reused.

**Pull request merged**
